### PR TITLE
DUPLO 17988 TF Provider crash: EKS credentials data source

### PR DESCRIPTION
**Pull request by Duplo-nikv**
## Overview

Changed conditional check logic for datasource `duplocloud_eks_credentials` and `duplocloud_gke_credentials`

## Summary of changes

This PR does the following:

- ...
- ...

## Testing performed
Could not reproduce the issue, change based on log attached to ticket
- [ ] Using unit tests
- [ ] Manually, on my local system
- [ ] Manually, on a remote test system

## Describe any breaking changes

- ...

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-17988 TF Provider crash: EKS credentials data source](https://app.clickup.com/t/8655600/DUPLO-17988)

File: duplocloud/data_source_duplo_eks_credentials.go [hunk 1]
```diff
@@ -59,10 +59,21 @@ func dataSourceEksCredentialsRead(d *schema.ResourceData, m interface{}) error {
...
+	if infra == nil {
+		return fmt.Errorf("no infra configuration for plan %s found", planID)
+	}
+
+	// Now we know infra is not nil, proceed with checks
+	if !infra.EnableK8Cluster && infra.Cloud != 2 {
 		return fmt.Errorf("no kubernetes cluster for this plan %s", planID)
+	} else if infra.Cloud == 2 {
+		// Check for AksConfig only if Cloud is 2 (relevant scenario)
+		if infra.AksConfig == nil {
```
Comment (tahir-duplo) [inline duplocloud/data_source_duplo_eks_credentials.go:72]: Can we refactor condition like  `infra.AksConfig == nil || !infra.AksConfig.CreateAndManage`

File: terraform copy.tfstate
Comment (tahir-duplo) [inline terraform copy.tfstate:1]: Remove this file.

File: terraform.tfstate copy.backup
Comment (tahir-duplo) [inline terraform.tfstate copy.backup:1]: Remove this file.

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**